packages/analytics-js/src/services/ErrorHandler/ErrorHandler.ts

```diff
@@ -53,21 +62,24 @@ class ErrorHandler implements IErrorHandler {
   attachErrorListeners() {
     if ('addEventListener' in (globalThis as typeof window)) {
       (globalThis as typeof window).addEventListener('error', (event: ErrorEvent | Event) => {
-        this.onError(event, undefined, undefined, undefined, 'unhandledException');
+        this.onError(event, undefined, undefined, undefined, ErrorType.UNHANDLEDEXCEPTION);
       });
 
       (globalThis as typeof window).addEventListener(
         'unhandledrejection',
         (event: PromiseRejectionEvent) => {
-          this.onError(event, undefined, undefined, undefined, 'unhandledPromiseRejection');
+          this.onError(event, undefined, undefined, undefined, ErrorType.UNHANDLEDREJECTION);
         },
       );
     } else {
       this.logger?.debug(`Failed to attach global error listeners.`);
     }
   }
 
-  init(externalSrcLoader: IExternalSrcLoader) {
+  init(httpClient: IHttpClient, externalSrcLoader: IExternalSrcLoader) {
+    this.httpClient = httpClient;
```

Comment:
Why is the HTTP client instance passed again here when it is already part of the constructor?

Reply:
Ok, then remove it from the constructor implementation.

Reply:
Done